src/common/src/array/bytes_array.rs

```diff
@@ -199,16 +211,20 @@ impl ArrayBuilder for BytesArrayBuilder {
         }
     }
 
-    fn append<'a>(&'a mut self, value: Option<&'a [u8]>) {
+    fn append_n<'a>(&'a mut self, n: usize, value: Option<&'a [u8]>) {
         match value {
             Some(x) => {
-                self.bitmap.append(true);
-                self.data.extend_from_slice(x);
-                self.offset.push(self.data.len())
+                self.bitmap.append_n(n, true);
+                for _ in 0..n {
+                    self.data.extend_from_slice(x);
```

Comment:
How about reserving capacity first?

Reply:
Sure. By the way, there is an opportunity for optimization. We can store the bytes only once and let the indexes point to the same location. But then we have to double the index array, one for start and one for length/end.

Reply:
Very intelligent! 😄